x-pack/legacy/plugins/apm/server/routes/services.ts

```diff
@@ -86,15 +86,48 @@ export const serviceNodeMetadataRoute = createRoute(() => ({
   }
 }));
 
+export const serviceMapAllRoute = createRoute(() => ({
+  path: '/api/apm/service-map/all',
+  params: {
+    query: t.intersection([
+      t.partial({ environment: t.string }),
+      uiFiltersRt,
+      rangeRt
+    ])
+  },
+  handler: async ({ context, request }) => {
+    if (!context.config['xpack.apm.serviceMapEnabled']) {
+      return new Boom('Not found', { statusCode: 404 });
+    }
+    const setup = await setupRequest(context, request);
+    const {
+      query: { environment }
+    } = context.params;
+    return getServiceMap({ setup, environment });
+  }
+}));
+
 export const serviceMapRoute = createRoute(() => ({
-  path: '/api/apm/service-map',
+  path: '/api/apm/service-map/{serviceName}',
   params: {
-    query: rangeRt
+    path: t.type({
+      serviceName: t.string
+    }),
+    query: t.intersection([
+      t.partial({ environment: t.string }),
+      uiFiltersRt,
+      rangeRt
+    ])
   },
-  handler: async ({ context }) => {
-    if (context.config['xpack.apm.serviceMapEnabled']) {
-      return getServiceMap();
+  handler: async ({ context, request }) => {
+    if (!context.config['xpack.apm.serviceMapEnabled']) {
+      return new Boom('Not found', { statusCode: 404 });
```

Comment:
`serviceMapEnabled` should be removed

Reply:
... or, are there any reasons to let users disable it? We don't allow this for any other features.

Reply:
yeah we can safely remove this config.

Reply:
after talking with folks, the consensus was to keep the config until FF

Reply:
what's the reason to wait until FF if we decided to do this?